kafka/client.go

```diff
@@ -472,18 +474,23 @@ func (client *Client) ReadTopic(name string, refreshMetadata bool) (Topic, error
 	}
 
 	if refreshMetadata {
-		log.Printf("[DEBUG] Refreshing metadata")
-		err := c.RefreshMetadata()
+		log.Printf("[DEBUG] Refreshing metadata for topic '%s'", name)
+		err := c.RefreshMetadata(name)
+
+		if err == sarama.ErrUnknownTopicOrPartition {
+			err := TopicMissingError{msg: fmt.Sprintf("%s could not be found", name)}
+			return topic, err
+		}
 		if err != nil {
-			log.Printf("[ERROR] Error refreshing metadata %s", err)
+			log.Printf("[ERROR] Error refreshing topic '%s' metadata %s", name, err)
 			return topic, err
 		}
 		err = client.extractTopics()
```

Comment:
this will clear knowledge on the client of other topics, correct?

Reply:
No. Data about the other topics is retained. Sarama has a cache (map) of topic data and only the topics passed to the RefreshMetadata method get updated in the cache.